subprojects/docs/src/docs/userguide/publishing_maven.adoc

```diff
@@ -96,74 +100,84 @@ Maven restricts `groupId` and `artifactId` to a limited character set (`[A-Za-z0
 
-++++
-<sample dir="maven-publish/javaProject" id="publishing_maven:example:repo-url-from-variable" title="Configuring repository URL based on project version">
-    <sourcefile file="build.gradle" snippet="repo-url-from-variable"/>
-</sample>
-++++
+=== Example: Configuring repository URL based on project version
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/maven-publish/javaProject/build.gradle[tag=repo-url-from-variable]
+----
 
-Similarly, you can use a <<build_environment, project or system property>> to decide which repository to publish to. The following example uses the release repository if the project property `release` is set, such as when a user runs `gradle -Prelease publish`:
+Similarly, you can use a <<build_environment.adoc#build_environment, project or system property>> to decide which repository to publish to. The following example uses the release repository if the project property `release` is set, such as when a user runs `gradle -Prelease publish`:
 
-++++
-<sample dir="maven-publish/javaProject" id="publishing_maven:example:repo-url-from-variable" title="Configuring repository URL based on project property">
-    <sourcefile file="build.gradle" snippet="repo-url-from-project-property"/>
-</sample>
-++++
+=== Example: Configuring repository URL based on project property
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/maven-publish/javaProject/build.gradle[tag=repo-url-from-project-property]
+----
 
 [[publishing_maven:install]]
-=== Publishing to Maven Local
+== Publishing to Maven Local
 
 For integration with a local Maven installation, it is sometimes useful to publish the module into the Maven local repository (typically at _$USER_HOME/.m2/repository_), along with its POM file and other metadata. In Maven parlance, this is referred to as 'installing' the module.
 
-The Maven Publish Plugin makes this easy to do by automatically creating a api:org.gradle.api.publish.maven.tasks.PublishToMavenLocal[] task for each api:org.gradle.api.publish.maven.MavenPublication[] in the `publishing.publications` container. The task name follows the pattern of `publish__PubName__PublicationToMavenLocal`. Each of these tasks is wired into the `publishToMavenLocal` aggregate task. You do not need to have `mavenLocal()` in your `publishing.repositories` section.
+The Maven Publish Plugin makes this easy to do by automatically creating a link:{javadocPath}/org/gradle/api/publish/maven/tasks/PublishToMavenLocal.html[PublishToMavenLocal] task for each link:{groovyDslPath}/org.gradle.api.publish.maven.MavenPublication.html[MavenPublication] in the `publishing.publications` container. The task name follows the pattern of `publish__PubName__PublicationToMavenLocal`. Each of these tasks is wired into the `publishToMavenLocal` aggregate task. You do not need to have `mavenLocal()` in your `publishing.repositories` section.
```

Comment:
`javadocPath` -> `groovyDslPath`

Reply:
Can't change "PublishToMavenLocal" here either unless you have a different link I can use.